commit f7a5930c3f0931340516587241a841b1c6b31c86
Author: Thomas Pöchtrager <t.poechtrager@gmail.com>
Date:   Sat Jul 5 11:07:48 2014 +0200

    fix LD_LIBRARY_PATH

diff --git cctools/configure.ac cctools/configure.ac
index 4ac2d69..c4f6c47 100644
--- cctools/configure.ac
+++ cctools/configure.ac
@@ -6,9 +6,10 @@ AC_CANONICAL_TARGET
 
 AC_ARG_PROGRAM
 
-#force compiler to clang
+# force compiler to clang
 CC=clang
 CXX=clang++
+CPP="clang -E"
 
 CFLAGS="$CFLAGS `cat tmp/cflags 2>/dev/null`"
 CFLAGS="$CFLAGS -isystem /usr/local/include -isystem /usr/pkg/include -O3"
@@ -16,9 +17,9 @@ CXXFLAGS="$CXXFLAGS -std=c++0x `cat tmp/cxxflags 2>/dev/null`"
 CXXFLAGS="$CXXFLAGS -isystem /usr/local/include -isystem /usr/pkg/include -O3"
 
 LDFLAGS="$LDFLAGS -L/usr/local/lib `cat tmp/ldflags 2>/dev/null`"
-export LD_LIBRARY_PATH="`cat tmp/ldpath 2>/dev/null`"
 
-CPP="clang -E"
+LD_LIBRARY_PATH="$LD_LIBRARY_PATH:`cat tmp/ldpath 2>/dev/null`"
+export LD_LIBRARY_PATH
 
 # FIXME 'subdir-objects' breaks dependency tracking
 AM_INIT_AUTOMAKE([1.10 no-define no-dist-gzip dist-bzip2 tar-ustar subdir-objects no-dependencies])
