commit 7083dddbb0f106d791d313829ea7dc45db90e375
Author: Thomas Pöchtrager <t.poechtrager@gmail.com>
Date:   Thu Sep 11 00:34:16 2014 +0200

    add a workaround for a glibc 2.20 bug

diff --git cctools/include/sys/cdefs.h cctools/include/sys/cdefs.h
new file mode 100644
index 0000000..ca78da5
--- /dev/null
+++ cctools/include/sys/cdefs.h
@@ -0,0 +1,20 @@
+/*
+ * Workaround for a GLIBC bug.
+ * https://sourceware.org/bugzilla/show_bug.cgi?id=14952
+ */
+
+#include_next <sys/cdefs.h>
+
+#ifdef __GLIBC__
+
+#ifndef __extern_inline
+# define __extern_inline \
+  extern __inline __attribute__ ((__gnu_inline__))
+#endif
+
+#ifndef __extern_always_inline
+# define __extern_always_inline \
+  extern __always_inline __attribute__ ((__gnu_inline__))
+#endif
+
+#endif /* __GLIBC__ */
