--- configure.ac	2013-09-23 14:13:38.000000000 +0200
+++ configure.ac	2013-10-23 21:30:12.000000000 +0200
@@ -205,13 +205,13 @@
 AC_CHECK_PROG(HAVE_LLVM_CONFIG, llvm-config, true, false)
 
 if test "x$HAVE_LLVM_CONFIG" = "xfalse"; then
-        AC_CHECK_PROG(HAVE_LLVM_CONFIG_32, llvm-config-3.2, true, false)
-        if test "x$HAVE_LLVM_CONFIG_32" = "xfalse"; then
-                AC_MSG_ERROR([llvm-config or llvm-config-3.2 not found on your system,please install related packages])
+        AC_CHECK_PROG(HAVE_LLVM_CONFIG_34, llvm-config-3.4, true, false)
+        if test "x$HAVE_LLVM_CONFIG_34" = "xfalse"; then
+                AC_MSG_ERROR([llvm-config or llvm-config-3.4 not found on your system,please install related packages])
         else
-	        LLVM_LIBDIR=`llvm-config-3.2 --libdir`
+	        LLVM_LIBDIR=`llvm-config-3.4 --libdir`
         	if test -f $LLVM_LIBDIR/libLTO.so;then
-                	LTO_LIBS="-L`llvm-config-3.2 --libdir` -lLTO"
+                	LTO_LIBS="-L`llvm-config-3.4 --libdir` -lLTO"
                 	LTO_DEFS="-DLTO_SUPPORT=1"
         	fi
         fi
