commit 86ee16d49f06c07fb7d208d6f082b4a63c077293
Author: Johannes Schickel <lordhoto [AT] gmail>
Date:   Fri Feb 12 14:51:14 2016 +0100

    Fix build on systems with glibc's qsort_r.

diff --git cctools/ld64/src/3rd/qsort_r.c cctools/ld64/src/3rd/qsort_r.c
index a514ad8..1bae250 100644
--- cctools/ld64/src/3rd/qsort_r.c
+++ cctools/ld64/src/3rd/qsort_r.c
@@ -28,7 +28,6 @@
  */
 
 #define I_AM_QSORT_R
-#define qsort_r qsort_r_local
 
 #include <sys/cdefs.h>
 
@@ -37,6 +36,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define qsort_r qsort_r_local
+
 #ifndef __FreeBSD__
 /* flsl.c */
 int
