diff --git cctools/configure.ac cctools/configure.ac
index 8b59f6b..3411e34 100644
--- cctools/configure.ac
+++ cctools/configure.ac
@@ -12,7 +12,7 @@ CXX=clang++
 
 CFLAGS="$CFLAGS `cat tmp/cflags 2>/dev/null`"
 CFLAGS="$CFLAGS -isystem /usr/local/include -isystem /usr/pkg/include -O3"
-CXXFLAGS="$CXXFLAGS -std=gnu++0x `cat tmp/cxxflags 2>/dev/null`"
+CXXFLAGS="$CXXFLAGS -std=c++0x `cat tmp/cxxflags 2>/dev/null`"
 CXXFLAGS="$CXXFLAGS -isystem /usr/local/include -isystem /usr/pkg/include -O3"
 
 LDFLAGS="$LDFLAGS -L/usr/local/lib `cat tmp/ldflags 2>/dev/null`"
diff --git cctools/ld64/src/ld/code-sign-blobs/memutils.h cctools/ld64/src/ld/code-sign-blobs/memutils.h
index 8fa48d6..4c97973 100644
--- cctools/ld64/src/ld/code-sign-blobs/memutils.h
+++ cctools/ld64/src/ld/code-sign-blobs/memutils.h
@@ -33,6 +33,9 @@
 #include <stdlib.h>
 #include <algorithm>
 
+#ifdef __GLIBCXX__
+using namespace __gnu_cxx;
+#endif
 
 //
 // Encapsulate these very sharp tools in a separate (ugly-named) namespace
