--- configure.ac	2013-11-13 19:16:27.023967416 +0100
+++ configure.ac	2013-11-13 19:16:40.323967379 +0100
@@ -35,18 +35,6 @@
 AC_CHECK_TOOL([HOST_RANLIB], [ranlib], :)
 AC_CHECK_TOOL([HOST_AR], [ar], :)
 
-#detect buggy unistd.h contains __block.
-buggy_unistd=`grep "\*__block," /usr/include/unistd.h`
-if test "$buggy_unistd"; then
-  echo "==========================================="
-  echo "Buggy unistd.h found"
-  echo "\"__block\" is a reserved word of clang compiler for blocks support."
-  echo "Please replace \"__block\" in /usr/include/unistd.h to \"__libc_block\""
-  echo "==========================================="
-  exit
-fi
-
-
 AC_CHECK_HEADERS([llvm-c/lto.h], [], [AC_MSG_ERROR([*** lto.h not found, Please install llvm dev packages according to your system])])
 AC_CHECK_HEADERS([openssl/md5.h], [], [AC_MSG_ERROR([*** md5.h not found, Please install ssl dev packages according to your system])])
 AC_CHECK_HEADERS([uuid/uuid.h], [], [AC_MSG_ERROR([*** uuid.h not found, Please install uuid dev packages according to your system])])
