diff --git a/gcc/config/darwin-driver.cc b/gcc/config/darwin-driver.cc
index 4235c619108..d9130863f0a 100644
--- a/gcc/config/darwin-driver.cc
+++ b/gcc/config/darwin-driver.cc
@@ -64,7 +64,7 @@ validate_macosx_version_min (const char *version_str)
 
   major = strtoul (version_str, &end, 10);
 
-  /* macOS 10, 11, 12, 13, 14, 15 and 26 are known.
+  /* macOS 10 to 15 and 26 upwards are known.
      clang accepts up to 99.  */
   if (major < 10 || major > 99)
     return NULL;
@@ -160,15 +160,16 @@ darwin_find_version_from_kernel (void)
   if (*version_p++ != '.')
     goto parse_failed;
 
-  /* Darwin25 saw a transition to macOS 26.  */
-  if (major_vers >= 25)
+  /* macOS 27 aligned version number with Darwin27.  */
+  if (major_vers >= 27)
     /* Apple clang doesn't include the minor version or the patch level
        in the object file, nor does it pass it to ld  */
+    asprintf (&new_flag, "%d.00.00", major_vers);
+  /* Darwin25 saw a transition to macOS 26.  */
+  else if (major_vers >= 25)
     asprintf (&new_flag, "%d.00.00", major_vers + 1);
   /* Darwin20 saw a transition to macOS 11.  */
   else if (major_vers >= 20)
-    /* Apple clang doesn't include the minor version or the patch level
-       in the object file, nor does it pass it to ld  */
     asprintf (&new_flag, "%d.00.00", major_vers - 9);
   else if (major_vers - 4 <= 4)
     /* On 10.4 and earlier, the old linker is used which does not
diff --git a/gcc/testsuite/gcc.dg/darwin-minversion-link.c b/gcc/testsuite/gcc.dg/darwin-minversion-link.c
index 55f7c7ea837..bf974c5bc08 100644
--- a/gcc/testsuite/gcc.dg/darwin-minversion-link.c
+++ b/gcc/testsuite/gcc.dg/darwin-minversion-link.c
@@ -21,6 +21,7 @@
 /* { dg-additional-options "-mmacosx-version-min=014.000.00 -DCHECK=140000" { target *-*-darwin23* } } */
 /* { dg-additional-options "-mmacosx-version-min=015.000.00 -DCHECK=150000" { target *-*-darwin24* } } */
 /* { dg-additional-options "-mmacosx-version-min=026.000.00 -DCHECK=260000" { target *-*-darwin25* } } */
+/* { dg-additional-options "-mmacosx-version-min=027.000.00 -DCHECK=270000" { target *-*-darwin27* } } */
 
 int
 main ()
