--- configure	2013-10-02 19:22:22.125610322 +0200
+++ configure	2013-10-02 19:23:08.833608564 +0200
@@ -7742,7 +7742,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcrypto  $LIBS"
+LIBS="-lcrypto -ldl  $LIBS -lm"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -7805,7 +7805,7 @@
 #define HAVE_LIBCRYPTO 1
 _ACEOF
 
-  LIBS="-lcrypto $LIBS"
+  LIBS="-lcrypto -ldl $LIBS -lm"
 
 else
   have_libcrypto="0"
